### ISSUE TITLE
`CS8620` reported on non-null argument passed to extension method with non-nullable params collection parameter

### ISSUE THREAD

**Issue by just-ero** [dotnet/roslyn#80024]
**Full issue information + repro project**:  
https://github.com/just-ero/roslyn-issues-repros/tree/main/5.0.0/009_NullabilityWarning_ParamsExtension

### Version

> [!NOTE]  
> **Roslyn**: 5.0.0-2.25412.5 (b0b21b61)  
> **.NET SDK**: 10.0.100-preview.7.25380.108  
>  
> **Visual Studio Code**: 1.103.2  
> **C# Extension**: 2.89.19  
> **C# Dev Kit Extension**: 1.50.20  

### Steps to Reproduce

1. Add the following code wherever syntactically applicable:
    ```cs
    class C;

    static class E
    {
        extension(C)
        {
            static void Extension(params C[] values) { }
        }

        static void M()
        {
            C.Extension(new C());
        }
    }
    ```
2. Observe `Argument of type 'C' cannot be used for parameter 'values' of type 'C[]' in 'void extension(C).Extension(params C[] values)' due to differences in the nullability of reference types.`

### Expected Behavior

The warning is not reported. The parameter expects non-nullable values and `new C()` is not nullable.

### Actual Behavior

It is.

**Comment by ero-qt** [comment]
Of note: this is not observed in normal extension methods, nor in non-extension static methods.

**Comment by RikkiGibson** [comment]
Will try to get this done today

**Comment by visose** [comment]
@RikkiGibson I'm running into this exact same issue in the release version of .NET 10.

**Comment by jskeet** [comment]
Also discussed here: https://stackoverflow.com/questions/79826396 - if the fix was expected to be in .NET 10 already, this should be reopened.

**Comment by RikkiGibson** [comment]
The fix for this bug hasn't made it into the release version of .NET 10 yet. I apologize for the inconvenience. I will update this issue when the fix makes it out into a stable release.

**Comment by anfomin** [comment]
The bug is still there in updated SDK 10.0.101...

**Comment by ristogod** [comment]
I find it quite odd that this has been known since August and here in December the issue still persists.

What SDK can we expect this to be in?

**Comment by mcurros2** [comment]
The issue shouldn't be closed. It's not fixed. Please review and reopen.

This still produces warning CS8620 when the extension is begin used.  SDK 10.0.101

`public class InvokeTest
{
    public void Test()
    {
        var t = new Test();
        t.TestMethod("Test");
    }
}

public class Test
{
}

public static class TestExtension
{
    extension(Test test)
    {
        public void TestMethod(params string[] args)
        {
        }
    }
}`

**Comment by Atulin** [comment]
I'm seeing the same issue on .NET 10

**Comment by marcominerva** [comment]
This issue is happening also with .NET SDK 10.0.102: https://github.com/dotnet/roslyn/issues/82008

**Comment by ero-qt** [comment]
@marcominerva Roslyn was not updated from 10.0.101 to 10.0.102 (check with `#error version` in your code). That means the fix has not shipped. https://github.com/dotnet/roslyn/pull/80433 has the 18.3 milestone, to give you some expectations. I think you can close your issue.

**Comment by aradalvand** [comment]
Why are we not getting this fix two months after the official release of .NET 10?! It makes no sense.

**Comment by Frulfump** [comment]
> [@marcominerva](https://github.com/marcominerva) Roslyn was not updated from 10.0.101 to 10.0.102 (check with `#error version` in your code). That means the fix has not shipped. [#80433](https://github.com/dotnet/roslyn/pull/80433) has the 18.3 milestone, to give you some expectations. I think you can close your issue.

When's 18.3 expected? The milestone has no expected due date

**Comment by marcominerva** [comment]
I am still experiencing the issue when using .NET SDK 10.0.103 (Visual Studio 18.3.1).

**Comment by Frulfump** [comment]
> I am still experiencing the issue when using .NET SDK 10.0.103 (Visual Studio 18.3.1).

Seems like the fix will be in the 2xx feature band unless backported
https://github.com/dotnet/roslyn/pull/80433#issuecomment-3798934789
